modules/raycast.lua

```diff
@@ -0,0 +1,81 @@
+local raycast = {}
+
+function raycast.upCast(obj,dist,offset,multi)
+    dist = dist or 1
+    offset = offset or 0
+    multi = multi or 1
+    local oPos = obj.getPosition()
+    local oBounds = obj.getBoundsNormalized()
+    local oRot = obj.getRotation()
+    local orig = Vector(oPos[1],oPos[2]+offset,oPos[3])
+    local siz = Vector(oBounds.size.x*multi,dist,oBounds.size.z*multi)
+    local orient = Vector(oRot[1],oRot[2],oRot[3])
+    local hits = Physics.cast({
+        origin       = orig,
+        direction    = Vector(0,1,0),
+        type         = 3,
+        size         = siz,
+        orientation  = orient,
+        max_distance = 0,
+        --debug        = true,
+    })
+    local hitObjects = {}
+    for _,v in pairs(hits) do
+        if v.hit_object ~= obj then table.insert(hitObjects,v.hit_object) end
+    end
+    return hitObjects
+end
+
+function raycast.upCastRay(obj,dist)
+    local hits = Physics.cast({
+        origin = obj.getBoundsNormalized().center,
+        direction = Vector(0,1,0),
+        max_distance = dist,
+        --debug = true,
+    })
+    local hitObjects = {}
+    for _,v in pairs(hits) do
+        if v.hit_object ~= obj and not obj.hasTag("Balanced") and not obj.hasTag("Thematic") then
+            table.insert(hitObjects,v.hit_object)
+        end
+    end
+    return hitObjects
+end
+
+function raycast.upCastPosSizRot(oPos,size,rot,dist,multi,tags)
```

Comment:
this shouldn't be called tags anymore since that has a new meaning in tts, `types` would probably be appropriate now